client/src/components/Rating/RatingStars.js

```diff
@@ -18,17 +25,35 @@ export const RatingStars = () => {
     setStarSet(newStarsSet)
   }
 
+  const averageMarkCalc = (mark, averageMark, marksNumber) => {
+    return (averageMark * marksNumber + mark) / (marksNumber + 1)
+  }
+
   const saveOnServer = () => {
     // console.log('Rating '+ ratingForSaving + " stars")
     // console.log('not Saved') //TODO: добавить запрос на сервер
   }
 
+  const ratingSaveHandler = async(body) => {
+    try{
+        const data = await request('/api/rating/save', 'POST', {...body})    ///TODO: вставить  правильный адрес
+    } catch(e) {
+    }
+  }
+
   const onStarClick = (e) => {
     e.preventDefault();
     const starId = e.currentTarget.id;
     changeStars(+starId);
-    setRating(+starId+1)
-    saveOnServer()
+    const currectAverage = Math.floor(averageMarkCalc(+starId, rating.averageRate, rating.totalMarks) * 10) / 10;
```

Comment:
why don't you put this `Math.floor` to `averageMarkCalc` function?

Reply:
Yeah, that's better, I didn't think about it...